omnigibson/object_states/toggle.py

```diff
@@ -51,16 +52,21 @@ def _initialize(self):
         pre_existing_mesh = get_prim_at_path(mesh_prim_path)
         # Create a primitive mesh if it doesn't already exist
         if not pre_existing_mesh:
-            button = UsdGeom.Sphere.Define(og.sim.stage, Sdf.Path(mesh_prim_path))
             self.radius = m.DEFAULT_RADIUS if self.radius is None else self.radius
+            mesh = create_primitive_mesh(
+                prim_path=mesh_prim_path,
+                primitive_type="Sphere",
+                extents=1.0,
+            )
         else:
             # Infer radius from mesh if not specified as an input
-            self.radius = pre_existing_mesh.GetAttribute("radius").Get() if self.radius is None else self.radius
+            recompute_extents(prim=pre_existing_mesh)
+            scale = np.array(pre_existing_mesh.GetAttribute("xformOp:scale").Get())
+            self.radius = np.cbrt(np.product(scale)) / 2.0 if self.radius is None else self.radius
```

Comment:
Sorry why do we need to do `np.cbrt(np.product(scale))`? Is this trying to be more general and to accommodate arbitrary meshes (that are not spheres)?